src/NzbDrone.Common/Http/HttpResponse.cs

```diff
@@ -11,25 +11,28 @@ public class HttpResponse
     {
         private static readonly Regex RegexSetCookie = new Regex("^(.*?)=(.*?)(?:;|$)", RegexOptions.Compiled);
 
-        public HttpResponse(HttpRequest request, HttpHeader headers, byte[] binaryData, HttpStatusCode statusCode = HttpStatusCode.OK)
+        public HttpResponse(HttpRequest request, HttpHeader headers, CookieCollection cookies, byte[] binaryData, HttpStatusCode statusCode = HttpStatusCode.OK)
         {
             Request = request;
             Headers = headers;
+            Cookies = cookies;
             ResponseData = binaryData;
             StatusCode = statusCode;
         }
 
-        public HttpResponse(HttpRequest request, HttpHeader headers, string content, HttpStatusCode statusCode = HttpStatusCode.OK)
+        public HttpResponse(HttpRequest request, HttpHeader headers, CookieCollection cookies, string content, HttpStatusCode statusCode = HttpStatusCode.OK)
         {
             Request = request;
             Headers = headers;
+            Cookies = cookies;
```

Comment:
Added CookieCollection to HttpResponse due to issue with arr Http Stack and getting cookies from Headers. Since they are stored in a dictionary in HttpResponse only the first cookie is ever available due to the limitation that there can only be one header key of a given value. Thus if the HttpWebRequest returns 3 cookies all of which will have header key of `Set-Cookie` only one is loaded in the dictionary. This is no good when we depend on storing cookies for Auth